sdk/core/azure-core/src/main/java/com/azure/core/util/AuthorizationChallengeHandler.java

```diff
@@ -103,6 +103,7 @@ public class AuthorizationChallengeHandler {
     private final Map<String, AtomicInteger> nonceTracker = new ConcurrentHashMap<>();
     private final AtomicReference<String> authorizationPipeliningType = new AtomicReference<>();
     private final AtomicReference<ConcurrentHashMap<String, String>> lastChallenge = new AtomicReference<>();
+    private final SecureRandom nonceGenerator = new SecureRandom();
```

Comment:
Use an instance-level nonce generator instead of creating a new one each time a nonce needs to be generated.